server/planning/types/planning.py

```diff
@@ -0,0 +1,113 @@
+from pydantic import Field
+from datetime import datetime
+from typing import Annotated, Any
+
+from content_api.items.model import CVItem, Place
+
+from superdesk.utc import utcnow
+from superdesk.core.resources import fields, dataclass
+from superdesk.core.resources.validators import validate_data_relation_async
+
+from .event import Translation
+from .base import BasePlanningModel
+from .common import RelatedEvent, Subject, PlanningCoverage
+from .enums import PostStates, UpdateMethods, WorkflowState
+
+
+@dataclass
+class Flags:
+    marked_for_not_publication: bool = False
+    overide_auto_assign_to_workflow: bool = False
+
+
+class PlanningResourceModel(BasePlanningModel):
+    guid: fields.Keyword
+    unique_id: fields.Keyword | None = None
+
+    firstcreated: datetime = Field(default_factory=utcnow)
+    versioncreated: datetime = Field(default_factory=utcnow)
+
+    # Ingest Details
+    ingest_provider: Annotated[fields.ObjectId, validate_data_relation_async("ingest_providers")] | None = None
+    source: fields.Keyword | None = None
+    original_source: fields.Keyword | None = None
+    ingest_provider_sequence: fields.Keyword | None = None
+    ingest_firstcreated: datetime = Field(default_factory=utcnow)
+    ingest_versioncreated: datetime = Field(default_factory=utcnow)
+
+    # Agenda Item details
+    agendas: list[Annotated[str, validate_data_relation_async("agenda")]] = Field(default_factory=list)
+    related_events: list[RelatedEvent] = Field(default_factory=list)
+    recurrence_id: fields.Keyword | None = None
+    planning_recurrence_id: fields.Keyword | None = None
+
+    # Planning Details
+    # NewsML-G2 Event properties See IPTC-G2-Implementation_Guide 16
+    # Planning Item Metadata - See IPTC-G2-Implementation_Guide 16.1
+    item_class: str = Field(default="plinat:newscoverage")
+    ednote: str | None = None
```

Comment:
I think this should be `fields.HTML`, same with `description_text` and `internal_note`, as these can contain rich text (aka HTML)